hal/src/main/native/sim/Interrupts.cpp

```diff
@@ -186,8 +186,8 @@ static int64_t WaitForInterruptDigital(HAL_InterruptHandle handle,
   interrupt->previousState = SimDIOData[digitalIndex].GetValue();
 
   int32_t uid = SimDIOData[digitalIndex].RegisterValueCallback(
-      &ProcessInterruptDigitalSynchronous, reinterpret_cast<void*>(dataHandle),
-      false);
+      &ProcessInterruptDigitalSynchronous,
+      reinterpret_cast<void*>(static_cast<size_t>(dataHandle)), false);
```

Comment:
What's the warning you're getting here?  Casting to size_t feels really strange.

Reply:
On 64 bit, there's a warning for reinterpret casting a 32 bit number to a 64 bit pointer. A static cast from 32 bit int to 64 bit pointer has an error, not a warning. So casting to size_t first, then to the pointer works. 

`warning C4312: 'reinterpret_cast': conversion from 'SynchronousWaitDataHandle' to 'void *' of greater size`

Reply:
Okay.  uintptr_t would be better IMO.

Reply:
Apparently I was doing void*->intptr_t->handle on the other end of this call, so I changed those to use uintptr_t as well.